front-end/src/components/main-page/TranscriptUploadModal.jsx

```diff
@@ -50,7 +50,18 @@ class TranscriptUploadModal extends React.Component {
   validateData(data) {
     try {
       const parsedData = JSON.parse(data);
-      return parsedData.data;
+      if (parsedData.data){
+        for (var i = 0; i < parsedData.data.length; i++) {
```

Comment:
Maybe change this to an enhanced for-loop?
```suggestion
        for (const intent of parsedData.data) {
```